src/Workspaces/Core/Portable/Diagnostics/DiagnosticDataSerializer.cs

```diff
@@ -65,11 +65,11 @@ public async Task<bool> SerializeAsync(IPersistentStorageService persistentServi
         private static string GetSerializationKeyForNonSourceDocument(TextDocument document, string key)
             => document.Id + ";" + key;
 
-        public async ValueTask<ImmutableArray<DiagnosticData>> DeserializeAsync(IPersistentStorageService persistentService, Project project, TextDocument? textDocument, string key, CancellationToken cancellationToken)
+        public async ValueTask<ImmutableArray<DiagnosticData>> DeserializeAsync(IPersistentStorageService2 persistentService, Project project, TextDocument? textDocument, string key, CancellationToken cancellationToken)
```

Comment:
📝 would prefer to see these still use `IPersistentStorageService` with the asynchronous method added there.